CHANGELOG.md

```diff
@@ -10,6 +10,7 @@
 - Python syntax highlighting no longer suffers from abysmal performance in specific scenarios. See #1688 (@keith-hall)
 - First line not shown in diff context. See #1891 (@divagant-martian)
 - Do not ignore syntaxes that handle file names with a `*.conf` extension. See #1703 (@cbolgiano)
+- Strips byte order mark from output in interactive mode. See #1922 (@dag-h)
```

Comment:
It's actually not dependent on interactive mode (atty). Because even when piping to a file, we want to strip the BOM if syntax highlighting is enabled.

I think the cases are:
 * In `config.loop_through` mode, never strip anything
 * When `config.colored_output` (I think it is that one) is enabled, always strip BOM, no matter if interactive or not. Otherwise regexes for syntaxes seems to break
 * If not loop through and if not colored output, I suppose we should keep the BOM, but maybe we can get away with stripping it if it makes the implementation much simpler

It would be fantastic if you could also ensure that there is a regression tests for each case above, although I would not go as far as saying that it is absolutely required for the PR.

Reply:
> If not loop through and if not colored output, I suppose we should keep the BOM

Actually, we should probably strip it also in this case, since some terminals have problems displaying it, and it doesn't do anything useful, especially not if it's in the "middle" of the output when bat prints its decorations.

Reply:
Sorry for the late response! I will have a look at this.

Reply:
Looks like this has been resolved with the three new regression tests.